superset/views/database/forms.py

```diff
@@ -90,7 +90,10 @@ def at_least_one_schema_is_allowed(database):
     csv_file = FileField(
         _("CSV File"),
         description=_("Select a CSV file to be uploaded to a database."),
-        validators=[FileRequired(), FileAllowed(["csv"], _("CSV Files Only!"))],
+        validators=[
+            FileRequired(),
+            FileAllowed(config["ALLOWED_EXTENSIONS"], _("CSV Files Only!")),
```

Comment:
Now that we're allowing TSV files as well, should this error message be updated?

Reply:
Good point; the supported extensions are now included in the error message. I also made the extension checking case-insensitive in `db_engine_specs/base.py` (guessing someone will trip on an upper-case CSV/TSV at some point).